sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -241,7 +243,8 @@ class SQLQueryTestSuite extends QueryTest with SharedSQLContext {
   private def listTestCases(): Seq[TestCase] = {
     listFilesRecursively(new File(inputFilePath)).map { file =>
       val resultFile = file.getAbsolutePath.replace(inputFilePath, goldenFilePath) + ".out"
-      TestCase(file.getName, file.getAbsolutePath, resultFile)
+      val absPath = file.getAbsolutePath
+      TestCase(absPath.stripPrefix(inputFilePath + File.separator), absPath, resultFile)
```

Comment:
@srowen I am adding a separator to make sure the test case name does not have a leading separator character. Example: 
```
file = /home/spark/sql/core/src/test/resources/sql-tests/inputs/subquery/exists-subquery/exists-basic.sql
inputFilePath = /home/spark/sql/core/src/test/resources/sql-tests/inputs
```
So i wanted the test case name to be `subquery/exists-subquery/exists-basic.sql` and not `/subquery/exists-subquery/exists-basic.sql`